kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js

```diff
@@ -0,0 +1,551 @@
+       * - data type - (text, int, float, workspaceObject (ref, name)
+       * - input app - input, select
+       */
+
+
+        // RENDERING
+
+        /*
+      The field widget is a generic wrapper around the input. It serves the following purposes:
+      - intercepts messages in order to display status.
+      appSpec - specifies the whole app
+      parameterSpec - just the segment of the appSpec that specifies this individual parameter
+      value - the initial value of this field
+      closeParameters - a list of "close" parameters, which might be context-dependent. E.g. for an output
+          field, this would be the list of all parameters meant to be output objects, so their names can
+          be cross-validated for uniqueness (optional)
+      */
+        function makeFieldWidget(appSpec, parameterSpec, value, closeParameters) {
+            return paramResolver.loadInputControl(parameterSpec)
+                .then(function (inputWidget) {
+                    let fieldWidget = FieldWidget.make({
+                        inputControlFactory: inputWidget,
+                        showHint: true,
+                        useRowHighight: true,
+                        initialValue: value,
+                        appSpec: appSpec,
+                        parameterSpec: parameterSpec,
+                        workspaceId: workspaceInfo.id,
+                        referenceType: 'name',
+                        paramsChannelName: paramsBus.channelName,
+                        closeParameters: closeParameters
+                    });
+
+                    // Forward all changed parameters to the controller. That is our main job!
+                    fieldWidget.bus.on('changed', function (message) {
+                        paramsBus.send({
+                            parameter: parameterSpec.id,
+                            newValue: message.newValue,
+                            isError: message.isError
+                        }, {
+                            key: {
+                                type: 'parameter-changed',
+                                parameter: parameterSpec.id
+                            }
+                        });
+
+                        paramsBus.emit('parameter-changed', {
+                            parameter: parameterSpec.id,
+                            newValue: message.newValue,
+                            isError: message.isError
+                        });
+                    });
+
+                    fieldWidget.bus.on('touched', function () {
+                        paramsBus.emit('parameter-touched', {
+                            parameter: parameterSpec.id
+                        });
+                    });
+
+
+                    // An input widget may ask for the current model value at any time.
+                    fieldWidget.bus.on('sync', function () {
+                        paramsBus.emit('parameter-sync', {
```

Comment:
What is the difference between `parameter-sync` and `sync-parameters`?

Reply:
i think this may be unnecessary as this chunk was copied over from the appCellWidget, which mixes instantiation of the file inputs, output and parameter fields. i didn't want to delete things as i'm still not sure what all the different behaviors we need for syncing data, but maybe @briehl has a better understanding